src/articles/_posts/2009-03-06-javascript-best-practices.md

```diff
@@ -546,26 +547,27 @@ As I am using the generic — really throw-away — variable names `ul` and `li`
 
 	function renderProfiles(o) {
 		var out = document.getElementById('profiles');
+		var ul = document.createElement('ul');
 		for(var i = 0; i < o.members.length; i++) {
-			var ul = document.createElement('ul');
 			var li = document.createElement('li');
-			li.appendChild(document.createTextNode(data.members[i].name));
-			li.appendChild(addMemberData(o.members[i]));
+			li.appendChild(document.createTextNode(o.members[i].name));
+			li.appendChild(addMemberData(o.members[i].data));
+			ul.appendChild(li);
 		}
 		out.appendChild(ul);
 	}
-	function addMemberData(member) {
+	function addMemberData(data) {
 		var ul = document.createElement('ul');
-		for(var i = 0; i < member.data.length; i++) {
+		for(var i = 0; i < data.length; i++) {
 			var li = document.createElement('li');
 			li.appendChild(
 				document.createTextNode(
-					member.data[i].label + ' ' +
-					member.data[i].value
+					data[i].label + ' ' +
+					data[i].value
 				)
 			);
+			ul.appendChild(li);
```

Comment:
Having `ul.appendChild(li);` outside the loop was similar to reassigning `ul` for every iteration. Only the last generated li node was being connected, all others ending up in the garbage collector.